### PR TITLE
fixes #9619 feat(nimbus): soft fail when fetches fail

### LINKED ISSUE #9619: Soft fail when fetches fail
We should not fail the entire fetch task if a single fetch fails. This is especially important as we start fetching versioned releases, which will result in a lot of fetches that could potentially fail.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3959)

### PR DESCRIPTION
**Pull request by brennie**
Because

- a single fetch failure currently breaks the entire fetch task; and
- we are about to drastically increase the number of fetches per task

This commit

- makes fetch failures not cause the tool to exit; and
- reports a summary of all successes and failures at the end of the task.

### CODE REVIEW

File: experimenter/manifesttool/fetch.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        print(f"fetch-latest: {app_name}: generate experimenter.yaml")
+        # The single-file fml file for each channel will generate the same
+        # experimenter.yaml, so we can pick any here.
+        nimbus_cli.generate_experimenter_yaml(
+            app_config,
+            channels[0],
+            manifest_dir,
+        )
+    except Exception as e:
```
Comment (jaredlockhart) [inline experimenter/manifesttool/fetch.py:65]: image

File: experimenter/manifesttool/fetch.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        for result in results:
+            if result.exc is None:
+                print(f"{result.app_name} at {result.ref_name} ({result.ref})")
+
+    print("")
+    if any(result.exc is not None for result in results):
+        print("FAILURES:\n")
+        for result in results:
+            if result.exc is not None:
+                print(f"{result.app_name} at {result.ref_name} ({result.ref})")
```
Comment (jaredlockhart) [inline experimenter/manifesttool/fetch.py:156]: Couldn't you just write this as

```py
        for result in results:
            if result.exc is None:
                print(f"SUCCESS: {result.app_name} at {result.ref_name} ({result.ref})")
            else:
                print(f"FAILED: {result.app_name} at {result.ref_name} ({result.ref})
```

File: experimenter/manifesttool/cli.py [hunk 3]
```diff
@@ -42,112 +39,16 @@ def fetch_latest(ctx: click.Context):
...
         context.manifest_dir.joinpath(app_config.slug).mkdir(exist_ok=True)
 
         if app_config.fml_path is not None:
-            fetch_fml_app(context, app_name, app_config)
+            results.append(fetch_fml_app(context.manifest_dir, app_name, app_config))
         elif app_config.experimenter_yaml_path is not None:
-            fetch_legacy_app(context, app_name, app_config)
+            results.append(fetch_legacy_app(context.manifest_dir, app_name, app_config))
         else:  # pragma: no cover
             assert False, "unreachable"
```
Comment (jaredlockhart) [inline experimenter/manifesttool/cli.py:52]: Wait why is this here?

**Review by jaredlockhart — APPROVED** [review]
Cool yeah good to handle this case now 👍

File: experimenter/manifesttool/cli.py (hunk 3 above)
Comment (brennie) [reply experimenter/manifesttool/cli.py:52]: We don't actually enforce that one of fml_path or experimenter_yaml_path is non none but in practice there is one or the other. Ill follow up to make it required in the model

File: experimenter/manifesttool/fetch.py (hunk 2 above)
Comment (brennie) [reply experimenter/manifesttool/fetch.py:156]: This groups all the successes and all the failures
